core/deployment/pom.xml

```diff
@@ -139,6 +139,10 @@
             <groupId>org.junit.jupiter</groupId>
             <artifactId>junit-jupiter</artifactId>
         </dependency>
+        <dependency>
+            <groupId>commons-io</groupId>
+            <artifactId>commons-io</artifactId>
```

Comment:
Why do you add this one here?

Reply:
commons-io was removed from maven-core, but it's still needed here because others deps required it, it was pulled transitively before but since it's excluded in many other dependencies that requires it then I added it explicitly.

In other words if I remove the declaration it will fail the tests.

Reply:
Maybe it's only needed in the test scope, I will need to check if that's the case.

Reply:
I think we need to have a better understanding of what needs commons-io. As I would rather get rid of it if it's not actually useful.
Note that we need to understand it better before moving it to a test dependency. As it could be required at runtime and then the tests would work but not the actual code.

Reply:
Hi @gsmet, the dependency is actually being used in `quarkus-core-deployment`:

```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.1:compile (default-compile) on project quarkus-core-deployment: Compilation failure
[ERROR] /quarkus/core/deployment/src/main/java/io/quarkus/deployment/steps/ConfigGenerationBuildStep.java:[30,29] package org.apache.commons.io does not exist
```

Its use is minimal, only for `FilenameUtils.removeExtension` and `FilenameUtils.getExtension`, so it may get refactored to not depend on commons-io, yet I think it's out of the scope of this PR.

Also, commons-io is being used in more places other than `quarkus-core-deployment`, so to completely remove the dependency from the project a bigger refactor should be done.

Reply:
OK, I think we probably need to get rid of it ASAP (and add some forbidden-apis rules) but I agree it's not related to this PR.